Libraries/Components/TextInput/TextInput.js

```diff
@@ -723,15 +720,47 @@ const TextInput = React.createClass({
     this.forceUpdate();
   },
 
+  _onSelectionChange: function(event: Event) {
+    this.props.onSelectionChange && this.props.onSelectionChange(event);
+
+    if (!this._inputRef) {
+      // calling `this.props.onSelectionChange`
+      // may clean up the input itself. Exits here.
+      return;
+    }
+
+    this._lastNativeSelection = event.nativeEvent.selection;
+
+    if (this.props.selection || this.props.selectionState) {
```

Comment:
I moved the call to this.props.selectionState.update to componentDidUpdate so we have to make sure the component updates on selection change. This ensures that selectionState is updated properly when the selection prop changes.